packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,198 @@
+import { includes, find, filter } from "lodash";
+import { XMLAttribute } from "@xml-tools/ast";
+import {
+  UI5SemanticModel,
+  UI5Class,
+} from "@ui5-language-assistant/semantic-model-types";
+import {
+  getUI5ClassByXMLElement,
+  getUI5AggregationByXMLElement,
+  isXMLNamespaceKey,
+  flattenProperties,
+  flattenEvents,
+  flattenAssociations,
+  flattenAggregations,
+  ui5NodeToFQN,
+} from "@ui5-language-assistant/logic-utils";
+import { UnknownAttributeKeyIssue } from "../../../api";
+import { TEMPLATING_NS, CUSTOM_DATA_NS } from "../../utils/special-namespaces";
+
+export function validateUnknownAttributeKey(
+  attribute: XMLAttribute,
+  model: UI5SemanticModel
+): UnknownAttributeKeyIssue[] {
+  if (attribute.syntax.key === undefined || attribute.key === null) {
+    // Can't give an error without a position or value
+    return [];
+  }
+
+  if (isAttributeNameAlwaysValid(attribute)) {
+    return [];
+  }
+
+  // Only validate (known) classes and aggregations
+  const xmlElement = attribute.parent;
+
+  // UI5 Class case
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
+  if (ui5Class !== undefined) {
+    if (isValidUI5ClassAttribute(ui5Class, attribute)) {
+      return [];
+    }
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  // UI5 Aggregation case
+  const ui5Aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  // Aggregations don't have additional allowed attributes
+  if (ui5Aggregation !== undefined) {
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  // No validations on unknown tags
+  return [];
+}
+
+function isAttributeNameAlwaysValid(attribute: XMLAttribute): boolean {
+  // This will never happen - it was checked before calling this function
+  /* istanbul ignore if */
+  if (attribute.key === null) {
+    return true;
+  }
+  // xmlns attributes are always valid
+  // TODO "xmlns:" should not return true (change when using new function in xml-tools)
+  if (isXMLNamespaceKey(attribute.key)) {
+    return true;
+  }
+
+  const { ns, name } = splitAttributeByNamespace(attribute);
+
+  // The following attributes are valid on all tags
+  // undefined name means all attribute names in the namespace are valid
```

Comment:
> // undefined name means all attribute names in the namespace are valid

Other options (instead of comment for special behavior/semantics)
- name: "*" (assuming * is not a valid name in XML)
- ```
     const ALL = undefined;
     const ALWAYS_VALID_ATTRIBUTES = [{ ns: CUSTOM_DATA_NS, name: ALL }]
  ```

Reply:
Done